cypress/integration/projectHealth/commits.ts

```diff
@@ -0,0 +1,257 @@
+    cy.dataCy("requester-select-options").should("not.exist");
+    cy.dataCy("commit-label").should("exist");
+    cy.dataCy("commit-label").each(($el) => {
+      cy.wrap($el).should("contain.text", "Git Tag");
+    });
+  });
+  describe("task filtering", () => {
+    beforeEach(() => {
+      cy.visit("/commits/spruce");
+    });
+    it("applying an `all` status filter should show all matching tasks with non failing tasks grouped", () => {
+      cy.dataCy("project-task-status-select").should("exist");
+      cy.dataCy("project-task-status-select").click();
+      cy.dataCy("project-task-status-select-options").should("be.visible");
+      cy.dataCy("project-task-status-select-options").within(() => {
+        cy.getInputByLabel("All").should("exist");
+        cy.getInputByLabel("All").should("not.be.checked");
+        cy.getInputByLabel("All").check({ force: true });
+      });
+      cy.dataCy("project-task-status-select").click();
+      cy.dataCy("project-task-status-select-options").should("not.be.visible");
+      cy.dataCy("grouped-task-status-badge").should("exist");
+      cy.dataCy("grouped-task-status-badge").should("have.length", 5);
+      cy.dataCy("waterfall-task-status-icon").should("exist");
+      cy.dataCy("waterfall-task-status-icon").should("have.length", 1);
+    });
+    it("applying a status filter should only show matching tasks", () => {
+      cy.dataCy("project-task-status-select").should("exist");
+      cy.dataCy("project-task-status-select").click();
+      cy.dataCy("project-task-status-select-options").should("be.visible");
+      cy.dataCy("project-task-status-select-options").within(() => {
+        cy.getInputByLabel("Succeeded").should("exist");
+        cy.getInputByLabel("Succeeded").should("not.be.checked");
+        cy.getInputByLabel("Succeeded").check({ force: true });
+      });
+      cy.dataCy("project-task-status-select").click();
+      cy.dataCy("project-task-status-select-options").should("not.be.visible");
+      cy.dataCy("grouped-task-status-badge").should("exist");
+      cy.dataCy("grouped-task-status-badge").should("have.length", 5);
+      cy.dataCy("grouped-task-status-badge").should(
+        "contain.text",
+        "Succeeded"
+      );
+      cy.dataCy("waterfall-task-status-icon").should("not.exist");
+    });
+    it("applying a build variant filter should show all task statuses by default", () => {
+      cy.getInputByLabel("Add New Build Variant Filter").type("Ubuntu");
+      cy.getInputByLabel("Add New Build Variant Filter").type("{enter}");
+      cy.dataCy("filter-badge").should("exist");
+      cy.dataCy("filter-badge").should("have.length", 1);
+      cy.dataCy("filter-badge").should("have.text", "buildVariants : Ubuntu");
+      cy.location("search").should("contain", "?buildVariants=Ubuntu");
+      cy.dataCy("grouped-task-status-badge").should("exist");
+      cy.dataCy("grouped-task-status-badge").should("have.length", 5);
```

Comment:
I kind of wonder if we need all of these `exist` assertions before these `have.length` assertions. I would think if the assertion `have.length` passes, then that would implicitly mean the element exists (and therefore there would be no need to test for it)

Reply:
When i wrote them i intended them to be a safe guard for trying to click on elements that may not exist yet. Since cypress tends to wait for elements to render when it makes assertions. I may have gone a little overboard :)